alerts/cluster-monitoring-operator/PrometheusOperatorRejectedResources.md

```diff
@@ -0,0 +1,89 @@
+$ ALERT_MANAGER=$(oc get route alertmanager-main -n openshift-monitoring -o jsonpath='{@.spec.host}')
+
+### Gets fired alerts
+$ curl -k -H "Authorization: Bearer $(oc create token prometheus-k8s -n openshift-monitoring)"  https://$ALERT_MANAGER/api/v1/alerts
+```
+
+### Diagnose using the Openshift web console
+
+1. Browse to **Observe** -> **Alerting**.
+2. Click the **Filter** drop-down list and select **Firing**.
+3. Search for the `PrometheusOperatorRejectedResources` alert.
+4. Click the alert to view its details.
+5. Scroll down to see the **Description** field. The namespace is displayed as
+a link, followed by the resource type.
+
+After you know the namespace and the resource type, you can look for a recently
+modified custom resource of that type in that namespace.
+**Note**: If the namespace is named `openshift-.*`, (excluding`openshift-user-workload-monitoring`),
+this might be a bug in OpenShift. Please file a support case.
+
+For more details about the error, view the logs for the Prometheus Operator pod in
+the `openshift-user-workload-monitoring` namespace.
+```bash
+# Get the name of the Prometheus Operator pod
+POD=$(oc get pod -n openshift-user-workload-monitoring | grep -Eo 'prometheus-operator-[0-9a-z]*-[0-9a-z]*')
+
+# Check its logs
+oc logs $POD -n openshift-user-workload-monitoring
+```
+
+The following is a typical error message about a rejected custom resource:
+```log
+level=warn ts=2023-07-03T20:37:20.740723141Z caller=operator.go:1917 component=prometheusoperator msg="skipping servicemonitor" error="it accesses file system via bearer token file which Prometheus specification prohibits" servicemonitor=quarkus-demo/otel-collector namespace=openshift-user-workload-monitoring prometheus=user-workload
+```
+
+## Mitigation
+
+Fix the misconfigured custom resource and reapply it to the cluster.
+
+The custom resource itself must already conform to the CRD schema. Otherwise,
+the API server will reject it before the Prometheus Operator does.
+
+This error often occurs because the Operator cannot apply a value from certain
+fields to create a valid configuration. The causes for this alert can vary
+depending on the specific configuration and deployment of Prometheus Operator.
+Possible causes include the following:
+
+- Violation of file system access rules through a bearer token file or
+a TLS certification file.
+- AlertManager:
+  - Invalid receiver settings--for example, a missing URL in Slack action.
+  - Invalid route settings.
+  - Settings request a feature that is unavailable in this version.
+  - Unsupported match rules in inhibition rules.
+- Prometheus & ServiceMonitor & PodMonitor:
+  - An invalid relabeling configuration--for example, a malformed regular expression.
+  - An invalid TLS configuration.
+  - A scrape interval configured to be longer than the scrape timeout.
+  - Information missing in authentication settings.
+  - A `PrometheusRules` object contains an invalid expression.
```

Comment:
this should be caught by the admission webhook service (except if the service is down).

Reply:
will mention this error should be caught by admission webhook.